hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileSystem.java

```diff
@@ -1655,12 +1653,10 @@ public void testTrash() throws Exception {
 
     // Call moveToTrash. We can't call protected fs.rename() directly
     trash.moveToTrash(path);
-    // Added this assertion here and will be tested as part of testTrash
-    // test case which needs to be tested with separate mini cluster having
-    // emptier thread started with close match of timings of relevant
-    // assertion statements and corresponding trash and checkpoint interval.
+
     Assert.assertTrue(o3fs.exists(userTrash));
-    Assert.assertTrue(o3fs.exists(userTrashCurrent));
+    Assert.assertTrue(o3fs.exists(userTrashCurrent) || o3fs.listStatus(
+        o3fs.listStatus(userTrash)[0].getPath()).length > 0);
```

Comment:
o3fs.exists(userTrashCurrent) this may not be required to check here as o3fs.listStatus(userTrash)[0].getPath()).length > 0 will be true even if it is in user trash current directory.

Reply:
The second condition is a chained list status, It is possible that the first listStatus returns Current and when we perform ls on current it returns key not found as checkpointing took place. We cannot guarantee checkpointing will happen as the clock doesn't start based on the time of key delete. but we can guarantee that deletion of checkpoint will happen atleast after keyDeletiontime + trashInterval.
If Current dir is present first condition returns true , it will not compute the second condition (short circuit operator) and only if first results false we check second.

Reply:
I missed the chained listStatus, that makes sense to ensure actual key is present.
But here we are ensuring Current directory exist but not checking for keys inside, as it might see key not found because checkpoint took place at the same time.
Can we wait here to ensure key is actually present either in Current or checkpoint directory?

Reply:
Changed it to check for presence of key in current or at least an entry inside checkpoint.

Reply:
Looks good now.